cpp/arcticdb/async/tasks.hpp

```diff
@@ -315,6 +291,113 @@ struct CopyCompressedTask : BaseTask {
     }
 };
 
+// Used in arcticdb-enterprise, do not remove without checking whether it is still used there
+struct CopyCompressedInterStoreTask : async::BaseTask {
+
+    struct AllOk {};
+
+    struct FailedTargets {
+        std::set<std::string, std::less<>> failed_targets_;
+    };
+
+    using ProcessingResult = std::variant<AllOk, FailedTargets>;
+
+    CopyCompressedInterStoreTask(entity::VariantKey key_to_read,
+                                 std::optional<entity::AtomKey> key_to_write,
+                                 bool check_key_exists_on_targets,
+                                 bool retry_on_failure,
+                                 std::shared_ptr<Store> source_store,
+                                 std::vector<std::shared_ptr<Store>> target_stores,
+                                 std::shared_ptr<BitRateStats> bit_rate_stats=nullptr)
+        : key_to_read_(std::move(key_to_read)),
+          key_to_write_(std::move(key_to_write)),
+          check_key_exists_on_targets_(check_key_exists_on_targets),
+          retry_on_failure_(retry_on_failure),
+          source_store_(std::move(source_store)),
+          target_stores_(std::move(target_stores)),
+          bit_rate_stats_(std::move(bit_rate_stats)){
+        ARCTICDB_DEBUG(log::storage(), "Creating copy compressed inter-store task from key {}: {} -> {}: {}",
+                       variant_key_type(key_to_read_),
+                       variant_key_view(key_to_read_),
+                       key_to_write_.has_value() ? variant_key_type(key_to_write_.value()) : variant_key_type(key_to_read_),
+                       key_to_write_.has_value() ? variant_key_view(key_to_write_.value()) : variant_key_view(key_to_read_));
+    }
+
+    ARCTICDB_MOVE_ONLY_DEFAULT(CopyCompressedInterStoreTask)
+
+    ProcessingResult operator()() {
+        auto res = copy();
+
+        if (!res.empty() && retry_on_failure_) {
+            res = copy();
+        }
+
+        if (!res.empty()) {
+            return FailedTargets{std::move(res)};
+        }
+
+        return AllOk{};
+    }
+
+private:
+    entity::VariantKey key_to_read_;
+    std::optional<entity::AtomKey> key_to_write_;
+    bool check_key_exists_on_targets_;
+    bool retry_on_failure_;
+    std::shared_ptr<Store> source_store_;
+    std::vector<std::shared_ptr<Store>> target_stores_;
+    std::shared_ptr<BitRateStats> bit_rate_stats_;
+
+    // Returns an empty set if the copy succeeds, otherwise the set contains the names of the target stores that failed
+    std::set<std::string, std::less<>> copy() {
```

Comment:
Why specifying `std::less<>`?

Reply:
And does it need to be ordered?

Reply:
The ordering seems just to be used for logging in enterprise so I'll make this unordered. The `std::less<>` was to resolve a Sonarlint warning that was probably spurious.